src/user-statuses/UserStatusScreen.js

```diff
@@ -36,74 +58,136 @@ type Props = $ReadOnly<{|
-    sendToServer({ status_text: statusText });
-  }, [statusText, sendToServer]);
+  const handlePressSave = useCallback(() => {
+    sendToServer({
+      status_text: statusTextFromInputValue(textInputValue),
+      status_emoji: statusEmojiFromInputValue(emojiInputValue),
+    });
+  }, [textInputValue, emojiInputValue, sendToServer]);
 
   const handlePressClear = useCallback(() => {
-    setStatusText(null);
-    sendToServer({ status_text: null });
-  }, [sendToServer]);
+    setTextInputValue(inputValueFromStatusText(null));
+    setEmojiInputValue(inputValueFromStatusEmoji(null));
+  }, []);
 
   return (
     <Screen title="User status">
-      <Input
-        autoFocus
-        maxLength={60}
-        style={styles.statusTextInput}
-        placeholder="What’s your status?"
-        value={statusText || ''}
-        onChangeText={setStatusText}
-      />
-      <FlatList
-        data={statusSuggestions}
-        keyboardShouldPersistTaps="always"
-        keyExtractor={item => item}
-        renderItem={({ item, index }) => (
-          <SelectableOptionRow
-            key={item}
-            itemKey={item}
-            title={item}
-            selected={item === statusText}
-            onRequestSelectionChange={itemKey => {
-              setStatusText(_(itemKey));
-            }}
+      <View style={styles.inputRow}>
+        {zulipFeatureLevel >= 86 && (
+          <EmojiInput
+            navigation={navigation}
+            value={emojiInputValue}
+            onChangeValue={setEmojiInputValue}
+            rightMargin
           />
         )}
-      />
-      <View style={styles.buttonsWrapper}>
-        <ZulipButton
-          style={styles.button}
-          secondary
-          text="Clear"
-          onPress={handlePressClear}
-          Icon={IconCancel}
-        />
-        <ZulipButton
-          style={styles.button}
-          text="Update"
-          onPress={handlePressUpdate}
-          Icon={IconDone}
+        <Input
+          autoFocus
+          maxLength={60}
+          style={styles.statusTextInput}
+          placeholder="What’s your status?"
+          value={textInputValue}
+          onChangeText={setTextInputValue}
         />
+        {(emojiInputValue !== null || textInputValue.length > 0) && (
+          <Pressable style={styles.clearButton} onPress={handlePressClear} hitSlop={12}>
+            {({ pressed }) => (
+              <Icon name="x" size={24} color={pressed ? HIGHLIGHT_COLOR : BRAND_COLOR} />
+            )}
+          </Pressable>
+        )}
       </View>
+      <FlatList
+        data={statusSuggestions}
+        keyboardShouldPersistTaps="always"
+        keyExtractor={(item, index) => index.toString() /* list is constant; index OK */}
+        renderItem={({ item: [emoji, text], index }) => {
+          const translatedText = _(text);
+          return (
+            <SelectableOptionRow
+              itemKey={index}
+              title={
+                zulipFeatureLevel >= 86
```

Comment:
Probably good to pull this comparison out as a local (of the render function, up by the selectors), like `const serverSupportsEmojiStatus = …`. Then that can get a TODO-server comment.